### PR TITLE
quantile matching

### PR DESCRIPTION
**Pull request by FelixWick**
In order to get a full individual probability distribution from a few quantile estimates, the CDF of an assumed probability distribution (or a smoothing spline for an unknown underlying distribution) is fitted to the quantile predictions.
Also, this fixes a few minor plotting issues introduced in recent updates.

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -429,6 +424,117 @@ def test_multiplicative_quantile_regression_90(is_plot, prepare_data, features,
...
+
+    i = 24
+    spl_fit_cdf = cdf_fit_spline(quantiles, cdf_values[:, i])
+
+    if is_plot:
+        plt.plot(quantiles, cdf_values[:, i], "ro")
+        xs = np.linspace(0, 1, 100)
+        plt.plot(xs, spl_fit_cdf(xs))
+        plt.savefig("spline_integration" + str(i) + ".png")
+        plt.clf()
```
Comment (lbventura) [inline tests/test_integration.py:453]: Shouldn't the test have an assert statement to guarantee that the generated CDF from the matching procedure yields the correct results? 

I might be overlooking something, but I fail to see how these tests add extra coverage beyond those defined in `test_quantile_matching.py`

File: tests/test_integration.py [hunk 2]
```diff
@@ -429,6 +424,117 @@ def test_multiplicative_quantile_regression_90(is_plot, prepare_data, features,
...
+
+
+@pytest.mark.skip(reason="Long running time")
+def test_multiplicative_quantile_regression_pdf_gamma(is_plot, prepare_data, features, feature_properties):
```
Comment (lbventura) [inline tests/test_integration.py:457]: Same as `test_multiplicative_quantile_regression_spline`

File: tests/test_integration.py [hunk 3]
```diff
@@ -429,6 +424,117 @@ def test_multiplicative_quantile_regression_90(is_plot, prepare_data, features,
...
+
+    cdf_truth = np.asarray(cdf_truth_list)
+    if is_plot:
+        plt.hist(cdf_truth[cdf_truth > 0], bins=30)
+        plt.savefig("gamma_cdf_truth_histo.png")
+        plt.clf()
+
+
+@pytest.mark.skip(reason="Long running time")
+def test_multiplicative_quantile_regression_pdf_nbinom(is_plot, prepare_data, features, feature_properties):
```
Comment (lbventura) [inline tests/test_integration.py:498]: Same as `test_multiplicative_quantile_regression_spline`

**Review by lbventura — APPROVED** [review]

File: tests/test_integration.py (hunk 1 above)
Comment (FelixWick) [reply tests/test_integration.py:453]: Yes, let me add some asserts :). The reason why I didn't include at first is that it is more a qualitative test here with the plots. And also a little bit of example how to use it.
The extra coverage of these tests is that we use the functionality on an actual training, i.e., quantile estimates with some uncertainty. That is a more difficult scenario than in the unit tests.

**Pull request merged**